src/test/kotlin/org/openmbee/mms5/BranchCreate.kt

```diff
@@ -0,0 +1,73 @@
+package org.openmbee.mms5
+
+import io.kotest.assertions.ktor.shouldHaveStatus
+import io.ktor.http.*
+import org.openmbee.mms5.util.*
+import java.util.*
+
+class BranchCreate : RefAny() {
+    init {
+        "reject invalid branch id" {
+            withTest {
+                httpPut("/orgs/$orgId/repos/$repoId/branches/bad branch id") {
+                    setTurtleBody(validBranchBodyFromMaster)
+                }.apply {
+                    response shouldHaveStatus HttpStatusCode.BadRequest
+                }
+            }
+        }
+
+        mapOf(
+            "rdf:type" to "mms:NotBranch",
+            "mms:id" to "\"not-$branchId\"",
+            "mms:etag" to "\"${UUID.randomUUID()}\"",
+            "mms:ref" to "<./nosuchbranch>"
+        ).forEach { (pred, obj) ->
+            "reject wrong $pred" {
+                withTest {
+                    httpPut(branchPath) {
+                        setTurtleBody("""
+                            <> dct:title "$branchName"@en .
```

Comment:
the request body also needs to specify a ref or commit otherwise each of these tests will fail before it even gets to the part that checks these properties
https://github.com/Open-MBEE/mms5-layer1-service/blob/develop/src/main/kotlin/org/openmbee/mms5/routes/BranchCreate.kt#L45-L54
https://github.com/Open-MBEE/mms5-layer1-service/blob/develop/src/main/kotlin/org/openmbee/mms5/MmsL1.kt#L245-L270